test/RandomSelector.jl

```diff
@@ -73,6 +73,18 @@
                 @test !any(isequal(2), diff(subset))
             end
         end
+
+        @testset "bad dates" begin
+            # Picking some random bad days
+            bad_dates = unique(rand(date_range, 10))
+            validation, holdout = partition(date_range, RandomSelector(42); bad_dates=bad_dates)
+
+            # Checking the bad dates are in neither set
+            is_bad(date) = in(date, bad_dates)
+            @test !any(is_bad.(validation))
+            @test !any(is_bad.(holdout))
```

Comment:
Not really a performance issue here, but in general best practice is:
```suggestion
            @test !any(is_bad, validation)
            @test !any(is_bad, holdout)
```
as it avoids allocations